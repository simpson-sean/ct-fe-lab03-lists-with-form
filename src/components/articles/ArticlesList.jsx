import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article'

const ArticleList = ({ articles }) => (
    <ul aria-label="articles">
        {articles?.map(({ title, author, description }, index) => (
            <li key={`${title}-${author}-${index}`}>
                <Article
                    title={title}
                    author={author}
                    description={description}
                />
            </li>
        ))}
    </ul>
);

ArticleList.propTypes = {
    articles: PropTypes.arrayOf(
        PropTypes.shape({
            title: PropTypes.string.isRequired,
            author: PropTypes.string.isRequired,
            description: PropTypes.string.isRequired
        })
    ).isRequired
};

export default ArticleList;