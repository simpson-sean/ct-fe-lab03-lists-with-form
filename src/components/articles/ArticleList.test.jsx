import React from 'react';
import { render, screen } from '@testing-library/react';
import ArticlesList from './ArticlesList';

describe('ArticlesList', () => {
  it('displays a list of Articles', () => {
    render(
      <ArticlesList
        articles={[
          {
            title: 'I like grapes',
            author: 'Farmer Dan',
            description: 'Grapes make wine',
          },
          {
            title: 'Ferrets make ok pets',
            author:'Guy Ferreto',
            description: 'They steal your things',
          },
        ]}
      />
    );

    const ul = screen.getByRole('list', { name: 'articles' });
    expect(ul).toMatchSnapshot();
  });
});