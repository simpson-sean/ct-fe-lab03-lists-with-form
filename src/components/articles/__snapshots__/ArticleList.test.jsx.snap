// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList displays a list of Articles 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <figure>
      <h1>
        I like grapes
      </h1>
      <h3>
        Farmer Dan
      </h3>
      <p>
        Grapes make wine
      </p>
    </figure>
  </li>
  <li>
    <figure>
      <h1>
        Ferrets make ok pets
      </h1>
      <h3>
        Guy Ferreto
      </h3>
      <p>
        They steal your things
      </p>
    </figure>
  </li>
</ul>
`;
